### PR TITLE
split Karaf feature into core and ext features

### PR DESCRIPTION
**Pull request by kaikreuzer**
All feature names stay the same, the only change here is to have two separate Maven artifacts for the ESH features. Everything under "extensions" has been moved to a new esh-ext artifact.

This makes it easier to provide updates of core and ext independent of each other - as discussed recently, this will help decoupling extensions from the core development.

I have also changed the artifactIds to not be called "karaf" anymore, but rather "esh-", because these names also appear in a Karaf-based solution.

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
yes, I thought about that option as well. If you prefer it, I will change the PR accordingly.

**Comment by kaikreuzer** [comment]
done.

File: features/karaf/esh-core/src/main/history/dependencies.xml [hunk 1]
```diff
@@ -0,0 +1,160 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (maggu2810) [inline features/karaf/esh-core/src/main/history/dependencies.xml]: Please change `features/karaf*/src/main/history` to `features/karaf/*/src/main/history` in `.gitignore` and don't add files that are placed in that directory.

File: features/karaf/esh-ext/src/main/history/dependencies.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (maggu2810) [inline features/karaf/esh-ext/src/main/history/dependencies.xml]: See above

File: features/karaf/esh-tp/src/main/history/dependencies.xml [hunk 3]
```diff
@@ -0,0 +1,113 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (maggu2810) [inline features/karaf/esh-tp/src/main/history/dependencies.xml]: See above

File: features/karaf/pom.xml [hunk 4]
```diff
@@ -1,35 +1,25 @@
...
   <parent>
     <groupId>org.eclipse.smarthome</groupId>
     <artifactId>features</artifactId>
     <version>0.9.0-SNAPSHOT</version>
   </parent>
 
-  <artifactId>karaf</artifactId>
-  <packaging>feature</packaging>
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.eclipse.smarthome</groupId>
```
Comment (maggu2810) [inline features/karaf/pom.xml]: If the groupId is equal to the parent one, there is no need to declare it explicitly (it is also okay to keep it, but not necessary).

**Comment by kaikreuzer** [comment]
Sorry, I should have paid more attention. Hope everything is fixed now.

**Pull request merged**